### PR TITLE
layers: Improve queue family checks

### LINKED ISSUE #1720: Layers are too restrictive with gotQueueFamilyPropertyCount for vkGetPhysicalDeviceSurfaceSupportKHR
In validation layers there is this [condition](https://github.com/KhronosGroup/Vulkan-LoaderAndValidationLayers/blob/572ea79/layers/swapchain.cpp#L266) (simplified here):

`if( qfp_count && !qfp_is_null ) gotQueueFamilyPropertyCount = true;`

I think this is over-restrictive. E.g. this (IMO valid and reasonable) made up code would generate an validation error:

    bool fun( VkPhysicalDevice physicalDevice, VkSurfaceKHR surface ){
        uint32_t qfCount;
        vkGetPhysicalDeviceQueueFamilyProperties( physicalDevice, &qfCount, nullptr );

        if( qfCount > 0 ){
            VkBool32 isSupported;
            VkResult errorCode = vkGetPhysicalDeviceSurfaceSupportKHR(
                physicalDevice, queueFamily, surface, &isSupported
            );

            return isSupported;
        }
        
       return false;
    }

> ERROR: PhysicalDevice(0x17523E6A540): 24, Swapchain, vkGetPhysicalDeviceSurfaceSupportKHR() called before calling the vkGetPhysicalDeviceQueueFamilyProperties function.

<hr/>

I can PR this myself, given time.

<hr/>

Also the `phy_data->numOfQueueFamilies` seems to be rewritten every time on `vkGetPhysicalDeviceQueueFamilyProperties`.

And `phy_data->gotQueueFamilyPropertyCount` seems to be reset every `vkEnumeratePhysicalDevices`.

That gives an Error for many valid usages of Vulkan, and I suspect for some "reasonable" usages too,

### PR DESCRIPTION
**Pull request by krOoze**
WIP &mdash; needs some testing.
known problems:
 - says "max value" instead "count"
 - crashes on `vkCreateDevice` if requesting queue family without details
 - Use `CALL_STATE` better
 - forgot skip in createSwapchain
 - checking should be in `parameter_validation`

<hr />

- migrate checks from swapchain layer
- make queue family count checks more permissive
- fixes #1720

### CODE REVIEW

File: layers/swapchain.cpp [hunk 1]
```diff
@@ -1401,19 +1132,21 @@ static PFN_vkVoidFunction intercept_khr_surface_command(const char *name, VkInst
 }
 
 static PFN_vkVoidFunction intercept_extension_instance_commands(const char *name) {
+    /*
```
Comment (mark-lunarg) [inline layers/swapchain.cpp]: Let's just delete this stuff rather than comment it out.

File: layers/core_validation.cpp [hunk 2]
```diff
@@ -3463,42 +3463,77 @@ VKAPI_ATTR void VKAPI_CALL DestroyInstance(VkInstance instance, const VkAllocati
     layer_data_map.erase(key);
 }
 
-// Verify that queue family has been properly requested
-static bool ValidateRequestedQueueFamilyProperties(instance_layer_data *instance_data, VkPhysicalDevice gpu,
-                                                   const VkDeviceCreateInfo *create_info) {
+static bool validate_queue_family_indices(instance_layer_data *instance_data, VkPhysicalDevice gpu,
```
Comment (mark-lunarg) [inline layers/core_validation.cpp]: Function names must be UpperCamelCase().  Also, all changes must be run through clang-format.

File: layers/vk_validation_error_database.txt [hunk 3]
```diff
@@ -56,12 +56,12 @@ VALIDATION_ERROR_00050~^~Y~^~Unknown~^~vkDestroyDevice~^~For more information re
...
-VALIDATION_ERROR_00058~^~N~^~Unknown~^~vkDestroyDevice~^~For more information refer to Vulkan Spec Section '4.3.2. Queue Creation' which states 'pQueuePriorities must be a pointer to an array of queueCount float values' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/html/vkspec.html#VkDeviceQueueCreateInfo)~^~implicit
-VALIDATION_ERROR_00059~^~N~^~Unknown~^~vkDestroyDevice~^~For more information refer to Vulkan Spec Section '4.3.2. Queue Creation' which states 'queueCount must be greater than 0' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/html/vkspec.html#VkDeviceQueueCreateInfo)~^~implicit
+VALIDATION_ERROR_00054~^~Y~^~Unknown~^~vkCreateDevice~^~For more information refer to Vulkan Spec Section '4.3.2. Queue Creation' which states 'queueFamilyIndex must be less than pQueueFamilyPropertyCount returned by vkGetPhysicalDeviceQueueFamilyProperties' (https://www.khronos.org/registry/vulkan/specs/1.0-extensions/html/vkspec.html#VkDeviceQueueCreateInfo)~^~
```
Comment (mark-lunarg) [inline layers/vk_validation_error_database.txt]: These changes were part of the last PR, right?  Might wanna rebase.

Comment (krOoze) [inline layers/vk_validation_error_database.txt]: Yes, already on it. I hoped GitHub is able to to do it seamlessly. Apparently not.

File: layers/core_validation.cpp (hunk 2 above)
Comment (krOoze) [inline layers/core_validation.cpp]: Right sorry,
Weird convention... I read only one rule and assumed underscores everywhere. But apparently Google likes some mixed naming scheme...

File: layers/swapchain.cpp (hunk 1 above)
Comment (krOoze) [inline layers/swapchain.cpp]: OK.

**Comment by krOoze** [comment]
I got distracted by the implementation of pre-existing facilities dealing with vkGetPhysicalDeviceQueueFamilyProperties outputs. I will PR some updates to that before I continue with this.

**Pull request closed without merging**